### PR TITLE
Added GetAppFiles endpoint on AppService

### PR DESCRIPTION
**Pull request by jstroem**

### CODE REVIEW

File: proto/toit/api/app.proto [hunk 1]
```diff
@@ -36,12 +37,31 @@ message CreateAppResponse {
...
+  }
+
+  bytes app_id = 1;
+  // if revision is zero will give the latest.
+  uint64 revision = 2;
+  string path = 3;
+  FileType type = 4;
+}
+
+message GetAppFilesResponse {
```
Comment (andersjohnsen) [inline proto/toit/api/app.proto:62]: no response? :D

**Review by andersjohnsen — APPROVED** [review]

File: proto/toit/api/app.proto [hunk 2]
```diff
@@ -36,12 +37,31 @@ message CreateAppResponse {
...
+  enum FileType {
+    UNKNOWN = 0;
+    DIRECTORY = 1;
+    FILE = 2;
+  }
+
+  bytes app_id = 1;
+  // if revision is zero will give the latest.
+  uint64 revision = 2;
+  string path = 3;
```
Comment (andersjohnsen) [inline proto/toit/api/app.proto:58]: type from path-ends-with-slash

File: proto/toit/api/app.proto [hunk 3]
```diff
@@ -36,12 +37,33 @@ message CreateAppResponse {
...
 
 message GetAppResponse {
   model.App app = 1;
 }
 
+message GetAppFilesRequest {
+  bytes app_id = 1;
+  // if revision is zero will give the latest.
+  uint64 revision = 2;
+  // if path ends with '/' when query on a directory.
```
Comment (andersjohnsen) [inline proto/toit/api/app.proto:52]: > // If path ends with '/', the request is for a directory.

File: proto/toit/api/app.proto [hunk 4]
```diff
@@ -36,12 +37,33 @@ message CreateAppResponse {
...
+  string path = 3;
+}
+
+message GetAppFilesResponse {
+  message Directory {
```
Comment (andersjohnsen) [inline proto/toit/api/app.proto:57]: Recursive? Directories?

**Review by andersjohnsen — APPROVED** [review]

File: proto/toit/api/app.proto [hunk 5]
```diff
@@ -36,12 +37,39 @@ message CreateAppResponse {
...
+  uint64 revision = 2;
+  // If path ends with '/', the request is for a directory.
+  string path = 3;
+}
+
+message GetAppFilesResponse {
+  message Entry {
+    string name = 1;
+    // content_type only set if not a directory
+    string content_type = 2;
```
Comment (andersjohnsen) [inline proto/toit/api/app.proto:60]: ENUM? Just to restrict it a bit, and then have an UNKNOWN_FILE or something...

**Review by andersjohnsen — APPROVED** [review]

File: proto/toit/api/app.proto (hunk 5 above)
Comment (jstroem) [reply proto/toit/api/app.proto:60]: since we want to save all kind of files the enum will be very long.

**Pull request merged**